manage-operational-layers/src/main/java/com/esri/arcgismaps/sample/manageoperationallayers/components/MapViewModel.kt

```diff
@@ -0,0 +1,173 @@
+        // get the index of the tapped layer
+        val layerIndex = operationalLayers.indexOf(operationalLayers.find { it.name == layerName })
+        // swap the selected layer with the layer on top
+        operationalLayers.swap(layerIndex, layerIndex - 1)
+        // update the layer names list
+        layerNames.apply {
+            clear()
+            addAll(operationalLayers.map { layer -> layer.name })
+        }
+        // update the operational layers
+        mapViewState.arcGISMap.operationalLayers.apply {
+            clear()
+            addAll(operationalLayers)
+        }
+    }
+
+    /**
+     * Swap the active layer with the layer on bottom.
+     */
+    fun moveLayerDown(layerName: String) {
+        // get a copy of the operational layers
+        val operationalLayers = mapViewState.arcGISMap.operationalLayers.toMutableList()
+        // if move down on the last item is selected, then return
+        if (operationalLayers.last().name == layerName) {
+            return
+        }
+        // get the index of the tapped layer
+        val layerIndex = operationalLayers.indexOf(operationalLayers.find { it.name == layerName })
+        // swap the selected layer with the layer on bottom
+        operationalLayers.swap(layerIndex, layerIndex + 1)
+        // update the layer names list
+        layerNames.apply {
+            clear()
+            addAll(operationalLayers.map { layer -> layer.name })
+        }
+        // update the operational layers
+        mapViewState.arcGISMap.operationalLayers.apply {
+            clear()
+            addAll(operationalLayers)
+        }
+    }
+
+    /**
+     * Removes [layerName] from map and adds it to the list of [inactiveLayers].
+     */
+    fun removeLayerFromMap(layerName: String) {
+        mapViewState.arcGISMap.operationalLayers.apply {
+            val layerIndex = indexOf(find { it.name == layerName })
+            inactiveLayers.add(get(layerIndex))
+            removeAt(layerIndex)
+            layerNames.removeAt(layerIndex)
+        }
+    }
+
+    /**
+     * Adds the [layerName] from the list of [inactiveLayers] to the map's operational layers.
+     */
+    fun addLayerToMap(layerName: String) {
+        inactiveLayers.apply {
+            val layerIndex = indexOf(find { it.name == layerName })
+            mapViewState.arcGISMap.operationalLayers.add(get(layerIndex))
+            layerNames.add(get(layerIndex).name)
+            removeAt(layerIndex)
+        }
+    }
+
+    /**
+     * Extension function to swap two values of a mutable list.
+     */
+    private fun <T> MutableList<T>.swap(index1: Int, index2: Int) {
+        val tmp = this[index1]
+        this[index1] = this[index2]
+        this[index2] = tmp
+    }
+}
+
+
+/**
+ * Class that represents the MapView state
+ */
+class MapViewState {
+    var arcGISMap: ArcGISMap by mutableStateOf(ArcGISMap(BasemapStyle.ArcGISTopographic))
+    var viewpoint: Viewpoint = Viewpoint(39.8, -98.6, 5e7)
```

Comment:
Is it worth making this observable? I see that it never changes so technically doesn't need to be but it might be misleading.